src/movies.js

```diff
@@ -1,28 +1,80 @@
 // Iteration 1: All directors? - Get the array of all directors.
-// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
-// How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  moviesArray = moviesArray.map((element) => element.director);
+
+  const uniqueMovArr = [];
+
+  for (let i = 0; i < moviesArray.length; i++) {
+    if (uniqueMovArr.indexOf(moviesArray[i]) === -1) {
+      uniqueMovArr.push(moviesArray[i]);
+    }
+  }
+  return uniqueMovArr
+}
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  if (moviesArray.length === 0) {return 0;}
+
+  return moviesArray.filter( (element) => element.director === 'Steven Spielberg')
```

Comment:
Here you can check 2 condition in one filter method ;)
`moviesArray.filter( (element) => element.director === 'Steven Spielberg' && element.genre.includes("Drama") )`